catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,568 @@
+      this.maxSideLength = maxSideLength;
+    } else {
+      LOGGER.debug(
+          "Invalid `maxSideLength` value [{}], must be greater than zero. Default value [{}] will be used instead.",
+          maxSideLength,
+          DEFAULT_MAX_SIDE_LENGTH);
+      this.maxSideLength = DEFAULT_MAX_SIDE_LENGTH;
+    }
+  }
+
+  public void setMaxNitfSizeMB(int maxNitfSizeMB) {
+    this.maxNitfSizeMB = maxNitfSizeMB;
+  }
+
+  public void setCreateOverview(boolean createOverview) {
+    this.createOverview = createOverview;
+  }
+
+  public void setStoreOriginalImage(boolean storeOriginalImage) {
+    this.storeOriginalImage = storeOriginalImage;
+  }
+
+  private void handleProcessCreateItem(List<ProcessCreateItem> processCreateItems) {
+    List<ProcessCreateItem> createdItems = new ArrayList<>();
+    processCreateItems.stream().map(this::handleProcessCreateItem).forEach(createdItems::addAll);
+    processCreateItems.addAll(createdItems);
+  }
+
+  private List<ProcessCreateItem> handleProcessCreateItem(ProcessCreateItem processCreateItem) {
+    List<ProcessCreateItem> createdItems = new ArrayList<>();
+    Metacard metacard = processCreateItem.getMetacard();
+    ProcessResource processResource = processCreateItem.getProcessResource();
+
+    String mimeType = processResource.getMimeType();
+
+    if (isNitfMimeType(mimeType)) {
+      if (processResource.getSize() / MEGABYTE <= maxNitfSizeMB) {
+        try (TemporaryFileBackedOutputStream fbos =
+            new TemporaryFileBackedOutputStream(DEFAULT_FILE_BACKED_OUTPUT_STREAM_THRESHOLD)) {
+
+          fbos.write(IOUtils.toByteArray(processResource.getInputStream()));
+          ByteSource byteSource = fbos.asByteSource();
+          BufferedImage renderedImage = renderImage(byteSource.openStream());
+
+          if (renderedImage != null) {
+            addThumbnailToMetacard(metacard, renderedImage);
+            processCreateItem.markMetacardAsModified();
+            if (createOverview) {
+              ProcessResource overviewProcessResource =
+                  createOverviewResource(renderedImage, metacard);
+              createdItems.add(new ProcessCreateItemImpl(overviewProcessResource, metacard));
+            }
+
+            if (storeOriginalImage) {
+              ProcessResource originalImageProcessResource =
+                  createOriginalImage(
+                      renderImageUsingOriginalDataModel(byteSource.openStream()), metacard);
+
+              createdItems.add(new ProcessCreateItemImpl(originalImageProcessResource, metacard));
+            }
+          }
+        } catch (IOException | NitfFormatException | RuntimeException e) {
+          LOGGER.debug(e.getMessage(), e);
+        } catch (InterruptedException e) {
+          LOGGER.error("Rendering failed for {}", processResource.getName(), e);
+        }
+      } else {
+        LOGGER.debug(
+            "Skipping large ({} MB) content item: name={}",
```

Comment:
✏️ `"Skipping content item (name={}, size={} MB) because it is larger than the configured maximum NITF file size to process of {} MB"`